chore: [autoapprove] Update `black` and `isort` to latest versions

holtskinner opened the pull request:
- Add Google Profile to `isort` `format` session
- Add `black[jupyter]` because several samples directories include jupyter notebooks

parthea commented on synthtool/gcp/templates/python_library/noxfile.py.j2:
```diff
@@ -179,7 +182,10 @@ def format(session):
     # See https://pycqa.github.io/isort/docs/configuration/options.html#force-sort-within-sections
     session.run(
         "isort",
+        "--profile=google",
         "--fss",
+        "--known-local-folder=main",
+        "--treat-all-comment-as-code",
```
This will cause a lot of changes in the generated libraries. Please create a doc to discuss the motivation for the change.

parthea replied:
The changes to this noxfile should also be in `synthtool/gcp/templates/python_mono_repo_library/noxfile.py.j2` for consistency between libraries in the split repositories vs monorepo.

holtskinner replied:
I'm going to revert this change, after experimentation, I realized it wouldn't improve the results.

parthea commented on synthtool/gcp/templates/python_library/noxfile.py.j2:
```diff
@@ -17,21 +17,23 @@
 
-BLACK_VERSION = "black==22.3.0"
-ISORT_VERSION = "isort==5.10.1"
+BLACK_VERSION = "black[jupyter]==23.3.0"
```
Please also update `black` here
https://github.com/googleapis/synthtool/blob/master/synthtool/gcp/templates/python_library/.pre-commit-config.yaml#L24

Can you clarify why the `jupyter` extra is needed?

holtskinner replied:
The jupyter extra is needed for repositories that contain jupyter notebooks like https://github.com/GoogleCloudPlatform/vertex-ai-samples/ and https://github.com/GoogleCloudPlatform/generative-ai

holtskinner replied:
Actually, on second thought. Should I add the `jupyter` extra to the python samples noxfile instead?

parthea reviewed (commented):
Added observations

holtskinner committed:
fix: Update `black` and `isort` to latest versions
- Add `black[jupyter]` because several samples directories include jupyter notebooks
- Add type hinting to avoid lint errors/warnings

holtskinner commented:
Cleaned up commit history.

parthea committed:
Merge branch 'master' into noxfile-update

parthea committed:
Merge branch 'master' into noxfile-update

holtskinner committed:
Merge branch 'master' into noxfile-update

holtskinner committed:
Merge branch 'master' into noxfile-update

holtskinner committed:
Merge branch 'master' into noxfile-update

holtskinner committed:
Merge branch 'master' into noxfile-update

parthea committed:
Merge branch 'master' into noxfile-update
